scripts/import-cp-contexts.sh

```diff
@@ -0,0 +1,247 @@
+# Define the echov function based on verbosity
+if [ "$arg_verbose" == "true" ]; then
+    echov() { >&2 echo "$@" ; }
+else
+    echov() { :; }
+fi
+
+
+# Determine the list of kubeconfigs to search
+# based on https://kubernetes.io/docs/reference/KUBECONFIG="$in_KUBECONFIG" kubectl/generated/KUBECONFIG="$in_KUBECONFIG" kubectl_config/
+echov "Determining the list of kubeconfigs to search:"
+if [[ "$arg_kubeconfig" != "" ]] ; then
+    in_KUBECONFIG="$arg_kubeconfig"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$arg_kubeconfig"
+    fi
+elif [[ "$KUBECONFIG" != "" ]] ; then
+    in_KUBECONFIG="$KUBECONFIG"
+    if [[ "$arg_output" == "" ]] ; then
+        if [[ "$KUBECONFIG" == *":"* ]] ; then
+            # hard to decide
+            # out_kubeconfig="$HOME/.kube/config"
+            >&2 echo "ERROR: please specify an output kubeconfig filename with \"-o <filename>\"!"
+            exit 1
+        else
+            out_kubeconfig="$KUBECONFIG"
+        fi
+    fi
+else
+    in_KUBECONFIG="$HOME/.kube/config"
+    if [[ "$arg_output" == "" ]] ; then
+        out_kubeconfig="$HOME/.kube/config"
+    else
+        out_kubeconfig="$arg_output"
+    fi
+fi
+echov "- input kubeconfigs: \"$in_KUBECONFIG\""
+echov "- output kubeconfig: \"$out_kubeconfig\""
+
+
+# Determine the list of contexts to search
+echov "Determining the list of contexts to search:"
+if [[ "$arg_context" == "" ]] ; then
+    contexts=($(KUBECONFIG="$in_KUBECONFIG" kubectl config get-contexts --no-headers -o name))
+else
+    contexts=("$arg_context")
+fi
+for context in "${contexts[@]}" ; do # for all contexts
+    echov "- found context \"${context}\""
+done
+
+
+echov "Getting kubeconfigs of KubeFlex Control Planes:"
+n=0
+for context in "${contexts[@]}" ; do # for all contexts
+    echov "- searching context \"${context}\""
+    cps=($(KUBECONFIG="$in_KUBECONFIG" kubectl --context "${context}" get controlplanes -no-headers -o name 2> /dev/null))
+    for i in "${!cps[@]}" ; do # for all control planes in context ${context}
+        name=${cps[i]##*/}
+        if [[ "$arg_names" != "" && ",$arg_names," != *",$name,"* ]] ; then
+            echov "  - skipping KubeFlex Control Plane \"$name\""
+            continue
+        fi
+        cp_name[n]=$name
+        cp_type[n]=$(KUBECONFIG="$in_KUBECONFIG" kubectl get controlplane ${cp_name[n]} -o jsonpath='{.spec.type}')
+        echov "  - found KubeFlex Control Plane \"${cp_name[i]}\" of type \"${cp_type[i]}\""
+        cp_kubeconfig[n]=$(get_kubeconfig "${context}" "${cp_name[n]}" "${cp_type[n]}" "$arg_ip_addr")
+        n=$((n+1))
+    done
+done
+
+
+if [[ "${#cp_name[@]}" == "0" ]] ; then
+    echov "No KubeFlex Control Planes found... nothing to do"
+    exit 0
+fi
+
+
+echov "Merging the contexts of KubeFlex Control Planes:"
+kubeconfig_list=""
+for i in "${!cp_name[@]}" ; do
+    echov "- \"${cp_name[i]}\" of type \"${cp_type[i]}\" ==> saving to temporary file \"$HOME/.kube/kubeconfig_${cp_name[i]}\""
```

Comment:
Why `~/.kube/` rather than the current working directory?

Reply:
ok